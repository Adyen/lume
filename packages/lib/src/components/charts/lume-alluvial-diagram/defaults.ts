import { sankeyJustify } from 'd3-sankey';

import { AlluvialDiagramOptions } from '@/composables/options';

export const options: AlluvialDiagramOptions = {
  margins: {
    top: 0,
    right: 0,
    bottom: 0,
    left: 0,
  },
  withAxes: false,
  withTooltip: false,
  withHover: false,
  withLegend: false,

  // Alluvial options
  gradient: false,
  highlightedElements: 'full',
  nodePadding: 16,
  nodeWidth: 16,
  nodeAlign: sankeyJustify,
  valueFormat: (value: number) => String(value),
};
