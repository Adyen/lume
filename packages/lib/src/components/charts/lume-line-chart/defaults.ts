import { DEFAULT_MARGINS } from '@/utils/margins';
import { ChartOptions } from '@/composables/options';

export const options: ChartOptions = {
  margins: DEFAULT_MARGINS.VERTICAL,
  startOnZero: true,
  withTooltip: true,
  xAxisOptions: {},
  yAxisOptions: { gridLines: true },
};
