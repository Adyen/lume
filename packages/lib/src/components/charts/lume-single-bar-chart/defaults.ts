import { ORIENTATIONS } from '@/utils/constants';
import { DEFAULT_MARGINS } from '@/utils/margins';

export const options = {
  [ORIENTATIONS.VERTICAL]: {
    withTooltip: true,
    xAxisOptions: {},
    yAxisOptions: { gridLines: true },
    margins: DEFAULT_MARGINS.VERTICAL,
  },
  [ORIENTATIONS.HORIZONTAL]: {
    withTooltip: true,
    xAxisOptions: { gridLines: true },
    yAxisOptions: { gridLines: false },
    margins: DEFAULT_MARGINS.HORIZONTAL,
  },
};
