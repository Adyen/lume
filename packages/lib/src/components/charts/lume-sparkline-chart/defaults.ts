import { LineChartOptions } from '@/composables/options';

export const options: LineChartOptions = {
  margins: {
    top: 0,
    right: 0,
    bottom: 0,
    left: 0,
  },
  withAxes: false,
  showArea: true,
  startOnZero: true,
  withLegend: false,
  tooltipOptions: {
    showTitle: false,
  },
};
