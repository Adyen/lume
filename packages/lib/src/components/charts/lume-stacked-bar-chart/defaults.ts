import { ORIENTATIONS } from '@/utils/constants';
import { DEFAULT_MARGINS } from '@/utils/margins';

export const options = {
  [ORIENTATIONS.VERTICAL]: {
    withTooltip: true,
    xAxisOptions: {},
    yAxisOptions: { gridLines: true },
    margins: DEFAULT_MARGINS.VERTICAL,
    tooltipOptions: { inverse: true },
  },
  [ORIENTATIONS.HORIZONTAL]: {
    withTooltip: true,
    xAxisOptions: { gridLines: true },
    yAxisOptions: { gridLines: false },
    margins: DEFAULT_MARGINS.HORIZONTAL,
    tooltipOptions: { position: 'right', inverse: true },
  },
};
