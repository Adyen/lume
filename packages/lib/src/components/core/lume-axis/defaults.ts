import type { AxisOptions } from '@/types/options';

const options: AxisOptions = {
  /** Displays a line for each tick. */
  gridLines: false,
  /** Displays the axis title. */
  withTitle: true,
  //
  //// Tick settings
  //
  /** Controls if the tick labels should be displayed. */
  showTicks: true,
  /** Amount of ticks to display in the axis. */
  tickCount: 10,
  /** Formatting string/function for the tick label. */
  tickFormat: null,
  /** Space between the tick label and the axis line. */
  tickPadding: 8,
};

export const xOptions: AxisOptions = {
  ...options,
  withTitle: false,
  skip: true,
};

export const yOptions: AxisOptions = {
  ...options,
  gridLines: true,
};
