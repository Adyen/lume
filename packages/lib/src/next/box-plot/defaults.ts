import { DEFAULT_MARGINS } from '@/utils/margins';

export const options = {
  xAxisOptions: {},
  yAxisOptions: { gridLines: true },
  margins: DEFAULT_MARGINS.VERTICAL,
};
