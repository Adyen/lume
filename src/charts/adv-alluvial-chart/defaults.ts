import { sankeyJustify } from 'd3-sankey';
import { DEFAULT_MARGINS } from '@/constants';

export const baseData = {
  values: [],
  nodePadding: 0,
  nodeWidth: 15,
  nodeAlign: sankeyJustify,
  valueFormatter: (value: number) => String(value),
};

export const options = {
  margins: DEFAULT_MARGINS.VERTICAL,
  showAxes: false,
  withTooltip: false,
  withHover: false,
  withLegend: false,
};

export const BASE_INSTANCE = {
  highlightedLink: null,
  highlightedNode: null,
  leftExtent: 0,
  rightExtent: 0,
  topExtent: 0,
  bottomExtent: 0,
  nodeBlocks: [],
  linkPaths: [],
};

export const transitionDuration = 200;
export const defaultChartColor = '03';
export const nodeToLabelGap = 6;
export const ghostStrokeWidthOffset = 25;
