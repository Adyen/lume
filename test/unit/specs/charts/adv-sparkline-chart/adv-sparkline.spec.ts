import { data, labels, xScale, yScale } from '../../mock-data';
import AdvSparkline from '@/charts/adv-sparkline-chart/adv-sparkline.vue';
import { options as defaultOptions } from '@/charts/adv-sparkline-chart/defaults';
import { BaseTestSuite } from "../../../reusable.test";
import { initiateCustomResizeObserverBeforeAll } from '../../../reusable.test';
import Vue from 'vue';

const sparklineChartTestSuiteFactory = (propsData) => new BaseTestSuite(AdvSparkline, propsData);

describe('adv-sparkline.vue', () => {
  const spy = initiateCustomResizeObserverBeforeAll();

  /*
   * NOTE: These first two tests are skipped, because we need a conainterSize update in order
   * to trigger the computed computedYScale property. We will return to the once that is up and running.
   * */
  test('mounts component and sets prop values', async () => {
    const wrapper = sparklineChartTestSuiteFactory({ data, labels, options: defaultOptions, xScale })
        .run()
        .wrapper;

    const el = wrapper.findComponent(AdvSparkline);
    const props = wrapper.props();

    // We need to trigger a resize for the computed properties to fall into shape
    el.trigger('resize');

    await Vue.nextTick();

    expect(el.exists()).toBe(true);

    expect(props).toHaveProperty('data');
    expect(props.data).toEqual(data);
    expect(props).toHaveProperty('labels');
    expect(props.labels).toEqual(labels);
    expect(props).toHaveProperty('options');
    expect(props.options).toEqual(defaultOptions);

    const areaPath = el.find('[data-j-sparkline__area]');
    expect(areaPath.exists()).toBe(true);
    expect(areaPath.classes().includes(`sparkline-chart__area--color-01`)).toBe(
      true
    );

    expect(wrapper.vm.$data).not.toHaveProperty('areaPathDefinition');
  });

  test('mounts component and sets custom area color', async () => {
    const areaColor = '02';
    const mutatedData = JSON.parse(JSON.stringify(data));
    mutatedData[0].areaColor = areaColor;

    const wrapper = sparklineChartTestSuiteFactory({ data: mutatedData, labels, xScale }).wrapper;

    const el = wrapper.findComponent(AdvSparkline);
    // We need to trigger a resize for the computed properties to fall into shape
    el.trigger('resize');

    await Vue.nextTick();

    const areaPath = el.find('[data-j-sparkline__area]');

    expect(
      areaPath.classes().includes(`sparkline-chart__area--color-${areaColor}`)
    ).toBe(true);
  });

  test('mounts component and checks areaPathDefinition', () => {
    const wrapper = sparklineChartTestSuiteFactory({ data, labels, xScale, yScale }).wrapper;

    const areaPathDefinition = (wrapper.vm as any).areaPathDefinition;
    expect(areaPathDefinition).toBeTruthy();
    expect(areaPathDefinition(xScale, null)).toBeFalsy();
    expect(areaPathDefinition(xScale, yScale)).toBeTruthy();
  });
});
