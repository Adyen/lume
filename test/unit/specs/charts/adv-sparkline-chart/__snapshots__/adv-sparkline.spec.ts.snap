// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adv-sparkline.vue mounts component and sets custom area color 1`] = `
<div
  class="adv-chart-container"
  data-j-adv-chart=""
  data-j-chart-container=""
  xscale="function scale(d) {
    var key = d + \\"\\", i = index.get(key);
    if (!i) {
      if (unknown !== implicit) return unknown;
      index.set(key, i = domain.push(d));
    }
    return range[(i - 1) % range.length];
  }"
>
  <div
    class="adv-chart__header"
  >
    <!---->
     
    <div
      class="adv-chart__controls"
    />
     
    <!---->
     
    <!---->
  </div>
   
  <svg
    class="adv-chart-container__svg"
    data-j-chart-container__root=""
    height="0"
  >
    <g
      class="adv-chart-container__group"
      data-j-chart-container__group=""
      transform="translate(0, 0)"
    />
  </svg>
   
  <!---->
   
  <!---->
   
</div>
`;

exports[`adv-sparkline.vue mounts component and sets prop values 1`] = `
<div
  class="adv-chart-container"
  data-j-adv-chart=""
  data-j-chart-container=""
  xscale="function scale(d) {
    var key = d + \\"\\", i = index.get(key);
    if (!i) {
      if (unknown !== implicit) return unknown;
      index.set(key, i = domain.push(d));
    }
    return range[(i - 1) % range.length];
  }"
>
  <div
    class="adv-chart__header"
  >
    <!---->
     
    <div
      class="adv-chart__controls"
    />
     
    <!---->
     
    <!---->
  </div>
   
  <svg
    class="adv-chart-container__svg"
    data-j-chart-container__root=""
    height="0"
  >
    <g
      class="adv-chart-container__group"
      data-j-chart-container__group=""
      transform="translate(0, 0)"
    />
  </svg>
   
  <!---->
   
  <!---->
   
</div>
`;
